kolibri/plugins/learn/assets/src/vue/toolbar/index.vue

```diff
@@ -0,0 +1,236 @@
+<template>
+
+  <div v-bind:class="['toolbar-show', displayToolBar ? 'toolbar-hide' : '' ]" >
+    <breadcrumbs class="breadcrumbs"></breadcrumbs>
+    <div :class="{ 'toggle-menu-on' : more }">
+      <label for="chan-select" :class="[ more ? 'lable-on' : 'visuallyhidden' ]" >Switch Channels</label>
+      <select
+        class="chan-select"
+        :class="[ more ? 'chan-select-mobile-location' : 'chan-select-location' ]"
+        id="chan-select"
+        name="chan-select"
+        v-model="getCurrentChannel"
+        @change="switchChannel($event)"
+      >
+        <option v-for="channel in getChannels" :value="channel.id">{{ channel.name }}</option>
+      </select>
+    </div>
+    <search-button @scrolling="handleScroll" class='search-btn'></search-button>
+    <button class="more" @click="toggleMore" ><svg src="../icons/more-ver.svg"></svg></button>
+  </div>
+
+</template>
+
+
+<script>
+
+  const constants = require('../../state/constants');
+  const getters = require('../../state/getters');
+
+  module.exports = {
+
+    data: () => ({
+      currScrollTop: 0,
+      delta: 5,
+      lastScrollTop: 0,
+      displayToolBar: false,
+      more: false,
+    }),
+    components: {
+      'search-widget': require('../search-widget'),
+      'search-button': require('../search-widget/search-button'),
+      'breadcrumbs': require('../breadcrumbs'),
+    },
+    computed: {
+      /*
+      * Get a list of channels.
+      */
+      getChannels() {
+        return this.channelList;
+      },
+      /*
+      * Get the current channel ID.
+      */
+      getCurrentChannel() {
+        return this.currentChannel;
+      },
+    },
+    methods: {
+      handleScroll(position) {
```

Comment:
This gets called _very_ often - would be best to put a throttle on how often the calculations below are performed.

See e.g. https://developer.mozilla.org/en-US/docs/Web/Events/scroll